src/main/java/com/cloudant/search3/Search.java

```diff
@@ -85,11 +86,11 @@ public void run() {
 
     private class CommitTask implements Runnable {
 
-        private final Subspace index;
+        private final SearchCacheKey key;
```

Comment:
since this was wrong previously, then none of the keys in the cache were being invalidated and then we were just populating it until it hit its max?

Reply:
it meamns we weren't ejecting them when they failed, but they'd be ejected (and closed) if they expired from the cache.